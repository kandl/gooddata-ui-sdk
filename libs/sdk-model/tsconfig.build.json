{
    "extends": "./tsconfig.json",
    "include": ["src/index.ts", "src/internal.ts"],
    "compilerOptions": {
        "baseUrl": null,
        "paths": null,
        "incremental": true,
        "declaration": true,
        "declarationMap": true
    }
}
