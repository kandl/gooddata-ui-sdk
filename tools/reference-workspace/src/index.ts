// (C) 2019-2021 GoodData Corporation

import * as ReferenceMd from "./md/full";
import * as ReferenceMdExt from "./md/ext";
import * as ReferenceRecordings from "./recordings";
import { DataSamples as ReferenceData } from "./recordings/dataSample";

export { ReferenceMd, ReferenceMdExt, ReferenceRecordings, ReferenceData };
