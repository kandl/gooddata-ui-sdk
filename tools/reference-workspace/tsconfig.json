{
    "compilerOptions": {
        "target": "es2017",
        "module": "NodeNext",
        "moduleResolution": "nodenext",
        "esModuleInterop": true,
        "lib": ["es2017", "dom"],
        "importHelpers": true,
        "outDir": "./esm",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "removeComments": false
    },
    "exclude": ["**/node_modules", "esm"]
}
