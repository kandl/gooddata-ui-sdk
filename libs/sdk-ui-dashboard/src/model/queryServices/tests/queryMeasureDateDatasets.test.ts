// (C) 2021-2022 GoodData Corporation
import { DashboardTester } from "../../tests/DashboardTester.js";
import { EmptyDashboardIdentifier } from "../../tests/fixtures/Dashboard.fixtures.js";
import { MeasureDateDatasets, queryDateDatasetsForMeasure } from "../../queries/index.js";
import { measureItem, ICatalogDateDataset } from "@gooddata/sdk-model";
import { initializeDashboard } from "../../commands/index.js";
import { ReferenceMd } from "@gooddata/reference-workspace";
import {
    MockAvailabilityWithDifferentRelevance,
    MockAvailabilityWithSameRelevance,
} from "../../tests/fixtures/CatalogAvailability.fixtures.js";

function datasetsDigest(
    datasets: ReadonlyArray<ICatalogDateDataset | undefined>,
): ReadonlyArray<string | undefined> {
    return datasets.map((d) => d?.dataSet.title);
}

describe("query measure date datasets", () => {
    describe("for datasets obtained from catalog", () => {
        // given all available datasets, this availability mock will pick 2 by name and associate relevance so that second one has more relevance

        it("should order date datasets by relevance desc", async () => {
            const Tester = DashboardTester.forRecording(
                EmptyDashboardIdentifier,
                {},
                {
                    catalogAvailability: {
                        availableDateDatasets: MockAvailabilityWithDifferentRelevance,
                    },
                },
            );

            await Tester.dispatchAndWaitFor(initializeDashboard(), "GDC.DASH/EVT.INITIALIZED");

            const result: MeasureDateDatasets = await Tester.query(
                queryDateDatasetsForMeasure(measureItem(ReferenceMd.Won)),
            );

            expect(datasetsDigest(result.dateDatasetsOrdered)).toMatchSnapshot();
            expect(result.dateDatasetDisplayNames).toMatchSnapshot();
        });

        it("should order date datasets by relevance and title if tied", async () => {
            const Tester = DashboardTester.forRecording(
                EmptyDashboardIdentifier,
                {},
                {
                    catalogAvailability: {
                        availableDateDatasets: MockAvailabilityWithSameRelevance,
                    },
                },
            );

            await Tester.dispatchAndWaitFor(initializeDashboard(), "GDC.DASH/EVT.INITIALIZED");

            const result: MeasureDateDatasets = await Tester.query(
                queryDateDatasetsForMeasure(measureItem(ReferenceMd.Won)),
            );

            expect(datasetsDigest(result.dateDatasetsOrdered)).toMatchSnapshot();
            expect(result.dateDatasetDisplayNames).toMatchSnapshot();
        });
    });
});
