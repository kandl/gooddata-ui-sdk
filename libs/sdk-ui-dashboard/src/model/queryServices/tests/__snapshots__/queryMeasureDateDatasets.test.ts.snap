// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`query measure date datasets > for datasets obtained from catalog > should order date datasets by relevance and title if tied 1`] = `
[
  "Date (Activity)",
  "Date (Timeline)",
]
`;

exports[`query measure date datasets > for datasets obtained from catalog > should order date datasets by relevance and title if tied 2`] = `
{
  "Date (Activity)": "Activity",
  "Date (Timeline)": "Timeline",
}
`;

exports[`query measure date datasets > for datasets obtained from catalog > should order date datasets by relevance desc 1`] = `
[
  "Date (Timeline)",
  "Date (Activity)",
]
`;

exports[`query measure date datasets > for datasets obtained from catalog > should order date datasets by relevance desc 2`] = `
{
  "Date (Activity)": "Activity",
  "Date (Timeline)": "Timeline",
}
`;
